src/Illuminate/Support/Str.php

```diff
@@ -251,6 +251,7 @@ public static function charAt($subject, $index)
      */
     public static function contains($haystack, $needles, $ignoreCase = false)
     {
+        $haystack ??= '';
```

Comment:
As specified on the PHP docs, the `$haystack` must be a string. If a piece of code is providing a `null`, instead of a `string`, the fix must be done at that level.